### PR TITLE
Add `RotationLike` type alias to the docs

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Add new type alias to the docs and update `TransformLike` as suggested in https://github.com/pyvista/pyvista/pull/6420#discussion_r1693894103.

EDIT: Note that the branch name should be `doc/rotationlike_alias` (not `boundslike`)

### CODE REVIEW

File: pyvista/core/_validation/validate.py [hunk 1]
```diff
@@ -544,7 +544,7 @@ def validate_transform3x3(transform, /, *, name="Transform"):
 
     Parameters
     ----------
-    transform : array_like | vtkMatrix3x3 | scipy.spatial.transform.Rotation
+    transform : RotationLike
```
Comment (user27182) [inline pyvista/core/_validation/validate.py:547]: `RotationLike` here implies that only 3x3 rotation matrices are supported. But any 3x3 matrix is supported... not sure how to communicate this, maybe just a note in the docs?

**Comment by user27182** [comment]
pre-commit.ci autofix

**Comment by user27182** [comment]
The new alias is properly configured 👍. It is automatically linked in docstrings, e.g.: https://66a6ca76975f4c9c9c977a9a--pyvista-dev.netlify.app/api/plotting/_autosummary/pyvista.prop3d.rotation_from

**Comment by user27182** [comment]
pre-commit.ci autofix

File: pyvista/core/_validation/validate.py (hunk 1 above)
Comment (tkoyama010) [reply pyvista/core/_validation/validate.py:547]: Yes, I think that will get the message to users.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Comment by user27182** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**